### ISSUE TITLE
French translation

### ISSUE THREAD

**Issue by mgeisler** [google/comprehensive-rust#323]

**Comment by mgeisler** [comment]
Hi @dblackstar, are you able to assign this issue to yourself?

**Comment by mgeisler** [comment]
Here is a Cloud Translate version of the `messages.po` file: [fr.zip](https://github.com/google/comprehensive-rust/files/10570187/fr.zip). I hope that can work as a good starting point.

**Comment by mgeisler** [comment]
> Oh yes it help a lot,

Excellent! Also, I just noted that there is a French translation of the Rust Book. That might be useful too: https://github.com/Jimskapt/rust-book-fr.

> i was figuring out how to host the translated book.

My plan is to publish the translations with the English version. We already publish a very incomplete Danish version here: https://google.github.io/comprehensive-rust/da/ (I'll add links to it when we have a more complete version).

**Comment by dblackstar** [comment]
>> Excellent! Also, I just noted that there is a French translation of the Rust Book. That might be useful too: https://github.com/Jimskapt/rust-book-fr.
>
>Okay
> 
> >My plan is to publish the translations with the English version. We already publish a very incomplete Danish version here: https://google.github.io/comprehensive-rust/da/ (I'll add links to it when we have a more complete version).
>
>Okay it's clear, i will just PR with the translated fr.po file

**Comment by ZoeThivet** [comment]
@dblackstar I'd be happy to help by reviewing the translation :)
I'm a native French speaker like you and I'm willing to provide an "inclusiveness" review so that the labels are gender neutral

**Comment by dblackstar** [comment]
> @dblackstar I'd be happy to help by reviewing the translation :) I'm a native French speaker like you and I'm willing to provide an "inclusiveness" review so that the labels are gender neutral

@ZoeThivet Yes sure it would be nice to work on that togother. Fork my Repo and for the guide on translation we follow https://github.com/Jimskapt/rust-book-fr

**Comment by mgeisler** [comment]
Hi @dblackstar and @ZoeThivet, I would very much encourage working together on this — there are nearly 30k words to translate so it's a big task!

The machine generated translation can perhaps help to get your started:

> Here is a Cloud Translate version of the `messages.pot` file: [fr.zip](https://github.com/google/comprehensive-rust/files/10570187/fr.zip).

**Comment by mgeisler** [comment]
> hey I'd like to help with the translation too, how can I participate? I'm a native french speaker

Hi @KookaS, that would be amazing!

I think @dblackstar is working in https://github.com/dblackstar/comprehensive-rust, but I don't see any new commits yet.

Please start by putting up a PR which adds a `po/fr.po` file which is mostly untranslated. Follow the instructions in https://github.com/google/comprehensive-rust/blob/main/TRANSLATIONS.md for that — and please let me know if there is anything unclear in them :smile:

**Comment by mgeisler** [comment]
The in-progress translation is now linked from the [translations page](https://google.github.io/comprehensive-rust/running-the-course/translations.html). Please comment here if you would like to help out!

**Comment by yohcop** [comment]
Hi! Just chiming in. I'll help the French translations as well!

**Comment by yohcop** [comment]
I found this page, in the french translation of the rust-book, which is quite handy:
https://jimskapt.github.io/rust-book-fr/translation-terms.html

I hope this helps!

**Comment by mgeisler** [comment]
Thanks @yohcop, that sounds awesome! I think the translation has stalled here, so any help is appreciated!

**Comment by mgeisler** [comment]
Hi everyone, I just wanted to let you know that there is a newly started French speaking Rust community on Reddit: https://www.reddit.com/r/rustfr/. Perhaps people there would have good input on how to translate different words in a nice way.
